# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/xfce4/xfce4-panel/isnull.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./panel/xfce_support.c.orig	2003-08-07 22:15:03.000000000 +0300
+++ ./panel/xfce_support.c	2003-11-09 16:55:51.000000000 +0200
@@ -955,10 +955,13 @@
 static gboolean
 delayed_exec(ActionCommand *command)
 {
-	real_exec_cmd(command->cmd, command->in_terminal, command->use_sn,
-			command->silent);
+	if (command->cmd != NULL)
+	{
+	    real_exec_cmd(command->cmd, command->in_terminal, command->use_sn,
+			    command->silent);
+	    g_free(command->cmd);
+	}
 
-	g_free(command->cmd);
 	g_free(command);
 
 	return(FALSE);
